frontend/src/components/common/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,52 @@
+import type { ComponentType, PropsWithChildren } from 'react';
+import { Component } from 'react';
+
+import { HTTPError } from '@api/HTTPError';
+
+import type { ErrorProps } from '../Error/Error';
+
+interface ErrorBoundaryProps {
+  Fallback: ComponentType<ErrorProps>;
+  onReset?: (error: Error | HTTPError) => void;
+}
+
+interface State {
+  hasError: boolean;
+  error: Error | null;
+}
+
+const initialState: State = {
+  hasError: false,
+  error: null,
+};
+
+class ErrorBoundary extends Component<PropsWithChildren<ErrorBoundaryProps>, State> {
+  state: State = initialState;
```

Comment:
```suggestion
const [errorState, setErrorState] = useState(initialState);
```
함수형으로 변경한다면 위와 같이 수정할 수 있을 것 같아요!